src/AzureExtension/DevBox/DevBoxInstance.cs

```diff
@@ -78,11 +80,13 @@ public class DevBoxInstance : IComputeSystem
         IDevBoxManagementService devBoxManagementService,
         IDevBoxOperationWatcher devBoxOperationWatcher,
         IDeveloperId developerId,
-        DevBoxMachineState devBoxMachineState)
+        DevBoxMachineState devBoxMachineState,
+        PackagesService packagesService)

```

Comment:
<non-blocking> We keep adding everything into IHost services and it seems like an abuse of dependency injection pattern. PackagesService is a utility class with one method that can be static.
It's unlikely it needs to be instantiated for a long period of time or we would create and destroy it often.
It's unlikely it will be mocked in a test (better be an interface in that case).
Its usage here is an implementation detail of DevBoxInstance class and I don't see why caller needs to know about that.
We can either just call that one static method or create an instance of PackagesService from DevBoxInstance whenever it's needed.
Even passing IHost or having access to it by other means seems like a better option to me than passing PackagesService

Reply:
While I agree we should be cautious, I don't believe what we have in the IHost is abusing it currently. We can remove the `IDevBoxAuthService devBoxAuthService,` now as its no longer used. I moved the use of it to the `DevBoxManagementService` class. After working with Host containers for a while now I've come to the realization that passing IHost isn't a good idea as that hides dependencies. Thats a small grip but on a bigger more established note looking at the .Net recommendations [here](https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-guidelines#recommendations) it says:

> Avoid using the [service locator pattern](https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-guidelines#scoped-service-as-singleton). For example, don't invoke [GetService](https://learn.microsoft.com/en-us/dotnet/api/system.iserviceprovider.getservice) to obtain a service instance when you can use DI instead.


if we see a class's dependencies becoming to big the right pattern is to move the dependencies to another class. Noted [here](https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-guidelines#design-services-for-dependency-injection):

> If a class has many injected dependencies, it might be a sign that the class has too many responsibilities and violates the [Single Responsibility Principle (SRP)](https://learn.microsoft.com/en-us/dotnet/standard/modern-web-apps-azure-architecture/architectural-principles#single-responsibility). Attempt to refactor the class by moving some of its responsibilities into new classes.

Overall I agree that this doesn't need the class that interacts with WinGet to be a service. Likely we'd probably want to either make a wrapper class over it over we can test it or we need an interface and a class that implements what you have in the package services class. This way we can mock it in a unit test

Reply:
I too believe we haven't overloaded the pattern ATM.
The idea was to make this an interface/class when I write the actual test. 
But I have changed it now anyway.

Reply:
It good with an interface. My concern is that we are creating singletons for every operations we may need only once and those singletons remain instantiated consuming device resources (yes, not much, but we keep adding them and it's hard to judge what resources are used by the underlying WinRT APIs). Consider making PackagesService a transient service instead of singleton.